test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -124,8 +134,58 @@
     private static DeepClone deepClone;
     //needed for @Nested
     private static Deque<Class<?>> currentTestClassStack = new ArrayDeque<>();
+    private static ScheduledExecutorService hangDetectionExecutor;
+    private static Duration hangTimeout;
+    private static ScheduledFuture<?> hangTaskKey;
+    private static final Runnable hangDetectionTask = new Runnable() {
+
+        final AtomicBoolean runOnce = new AtomicBoolean();
+
+        @Override
+        public void run() {
+            if (!runOnce.compareAndSet(false, true)) {
+                return;
+            }
+            System.err.println("@QuarkusTest has detected a hang, as there has been no test activity in " + hangTimeout);
```

Comment:
I wonder if it would be useful to be able (config?) to dump the stack trace to a file instead?

Reply:
Given that this is most likely going to be useful in CI, it is so much easier to just use system err. 
If we were to dump it to a file, it would need to be in a location that is then included in the artifacts produced by the CI pipeline

Reply:
Yeah, I'm talking about local builds... but the truth is that you can easily dump the stack manually in this case.

Reply:
Exactly